.ci/scripts/test.sh

```diff
@@ -11,54 +11,166 @@
 #     /bin/bash -c ".ci/scripts/docker-test.sh"
 #
 # Usage:
-#     .ci/scripts/tests.sh NODE_VERSION [TAV_MODULE] [IS_EDGE]
+#     .ci/scripts/tests.sh [-b release|rc|nightly] [-f] [-t TAV_MODULE] NODE_VERSION
 #
 # - NODE_VERSION is a version of Node.js, e.g. "14", usable both to select
 #   a image from Docker Hub:
 #       docker pull docker.io/library/node:$NODE_VERSION
 #   and to install Node with nvm:
 #       nvm install $NODE_VERSION
-# - TAV_MODULE, if specified and not the empty string, will result in
-#   "test-all-versions" (TAV) tests being run. It identifies the name of a
-#   module in ".tav.yml", e.g. "redis" or "pg". If this argument is not given
-#   then the regular agent tests (i.e. the same as 'npm test') will be run.
-# - IS_EDGE is "true" or "false" (the default). If true, then the node version
-#   used for testing will be installed via:
-#       NVM_NODEJS_ORG_MIRROR=${NVM_NODEJS_ORG_MIRROR} nvm install ${NODE_VERSION}
-#   where typically the mirror URL is also set to one of:
-#       https://nodejs.org/download/nightly/
-#       https://nodejs.org/download/rc/
-#   to test against a recent pre-releases of node.
+# - TAV_MODULE, if specified, will result in "test-all-versions" (TAV) tests
+#   being run. It identifies the name of a module in ".tav.yml", e.g. "redis" or
+#   "pg". If this argument is not given then the regular agent tests (i.e. the
+#   same as 'npm test') will be run.
+# - The "-b" option (BUILD_TYPE) can be used to test with pre-release "rc" or
+#   "nightly" builds of Node.js. This will skip out if there is already a
+#   release build for the given version.
+# - The "-f" option (FORCE) can be used to force a "nightly" or "rc" test,
+#   even if there is already a release build for the same version.
 #
 # Examples:
-#     .ci/scripts/test.sh 14                # regular tests against latest docker:14 image
-#     .ci/scripts/test.sh 14 "" false       # same
-#     .ci/scripts/test.sh 8.6               # regular tests against latest docker:8.6 image
-#     .ci/scripts/test.sh 14 "redis" false  # redis TAV tests against latest docker:14 image
-#     NVM_NODEJS_ORG_MIRROR=https://nodejs.org/download/nightly/ \
-#       .ci/scripts/test.sh 14 "" true      # regular tests against latest node 14 nightly release
+#     .ci/scripts/test.sh 14                # run agent tests against latest docker:14 image
+#     .ci/scripts/test.sh -b release 14     # same
+#     .ci/scripts/test.sh 8.6               # run agent tests against latest docker:8.6 image
+#     .ci/scripts/test.sh -t redis 14       # run redis TAV tests against latest docker:14 image
+#     .ci/scripts/test.sh -b nightly 17     # run agent tests against the latest v17 node.js nightly build
+#     .ci/scripts/test.sh -f -b nightly 17  # ... same, but force doing tests even if there
+#                                           #     is already a node.js v17 release
 #
 
-set -exo pipefail
+set -eo pipefail
```

Comment:
Presuming -x has been removed to make this all a bit less chatty.  This is a six of one, half dozen of the other decision for me -- just noting it out loud.

Reply:
It was added again below after arg processing:

```
# Turn on xtrace output only after processing args.
set -x
```

because otherwise for a simple `.ci/scripts/test.sh -h` to get help output, it resulted in lots of noise.


If we were over-engineering this, I personally typically have a starter Bash script block like this:

```bash
if [ "$TRACE" != "" ]; then
    export PS4='${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
    set -o xtrace
fi
set -o errexit
set -o pipefail
```

which allows full xtrace with the "TRACE=1" envvar. (The PS4 also gives detail like bash function and line number for each logged line.) We could do that at some point, if helpful.